### ISSUE TITLE
Microsoft.CodeAnalysis.NetAnalyzers do not find issues in VSTO Excel Add-in

### ISSUE THREAD

**Issue by ShaunLoganOracle** [dotnet/roslyn#77609]
**Version Used**: 
Visual Studio 2022 17.12.5
Microsoft.CodeAnalysis.NetAnalyzers 9.0.0
.NET Framework 4.8

**Steps to Reproduce**:

1. Create a VSTO Excel Add-in using VS22 Create Project wizard's "VSTO Excel Add-in" project template
2. Install NuGet package: Microsoft.CodeAnalysis.NetAnalyzers (packagereference)
3. Enable code analysis ("Microsoft All Rules" ruleset) by adding to CSPROJ file:
```
<CodeAnalysisRuleSet>AllRules.ruleset</CodeAnalysisRuleSet>
 <EnableNETAnalyzers>true</EnableNETAnalyzers>
 <AnalysisLevel>latest</AnalysisLevel>
```

4. Add some methods/properties that should general CA warnings (sample code provided on demand)
5. From VS22 menu, run Analyze > Run Code Analysis > On Solution
 
**Expected Behavior**:
I expect to see CA warnings generated for using `out object foo` in a method argument (among others);

**Actual Behavior**:
 the Output Window and Error List window don't show any CA warnings

**Other Notes**
I am trying to use this NuGet package CA approach since FxCopCmd stopped working in VS22 17.10.x (approx).  See [GH thread](https://github.com/dotnet/roslyn/issues/74228#issuecomment-2654750680) #74228 

I do see %userprofile%Microsoft.CodeAnalysis.NetAnalyzers\9.0.0\<many files> so I know that the package is accessible.
PackageReference in CSPROJ file:
```
  <ItemGroup>
    <PackageReference Include="Microsoft.CodeAnalysis.NetAnalyzers">
      <Version>9.0.0</Version>
      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
      <PrivateAssets>all</PrivateAssets>
    </PackageReference>
  </ItemGroup>
```

**Comment by ShaunLoganOracle** [comment]
@arunchndr , @sharwell -
Thanks for any attention you can give this issue - we are currently blocked from using Code Analysis for our add-in solution.  

I see that this bug was added to the **.NET 10** milestone.  If that means a fix will be considered for the next release (.NET 10), then that won't work for us as we have to use **.NET Framework** for our VSTO add-in.  If that milestone means something else, then fine.

**Comment by JoeRobich** [comment]
@ShaunLoganOracle I was unable to reproduce your issue. Does the following work for you? 

These are the steps I followed to reproduce your issue.
 - Created a Excel Addin using the template. 
 - Added the Microsoft.CodeAnalysis.NetAnalyzers using Package Manager
 - copied the AllRulesEnabled.ruleset from packages\Microsoft.CodeAnalysis.NetAnalyzers.9.0.0\rulesets to my project folder
 - Edited the .csproj to include the following
```xml
    <EnableNETAnalyzers>true</EnableNETAnalyzers>
    <CodeAnalysisRuleSet>AllRulesEnabled.ruleset</CodeAnalysisRuleSet>
```

**Comment by ShaunLoganOracle** [comment]
Thanks @JoeRobich 
With a quick trial using a simple add-in and things seem promising.  I'll do a deeper dive of our production code and reply back here, "shortly".

**Comment by ShaunLoganOracle** [comment]
Things are looking promising against our production code base, so closing this issue. I think the root cause of our problems was mismatched namespace in the ruleset file.
I'll open new issues for anything else we run into when we fully deploy the .NET Analyzers for our add-in's codebase,
